bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerOpenOp.java

```diff
@@ -174,9 +195,21 @@ private void openWithMetadata(Versioned<LedgerMetadata> versionedMetadata) {
         }
 
         // get the ledger metadata back
+        final boolean watchImmediately = !doRecovery || metadata.isClosed();
```

Comment:
We changed the behavior here. When the replication worker opens ledgers without doRecovery, it may check all the ledgers in the BookKeeper cluster. What's more, most of the ledgers are in a closed state. It will register a lot of listeners and bring heavy pressure on Zookeeper.